packages/costa/src/runtime.ts

```diff
@@ -247,22 +246,22 @@ export class CostaRuntime {
   async fetch(name: string): Promise<PluginPackage> {
     let pkg: PluginPackage;
     const source = this.getSource(name);
-    if (source.from === 'npm') {
+    if (source.protocol === 'npm') {
```

Comment:
May I ask you not to change the field name?

Reply:
I think the field name is not very accurate. IMO, `protocol` + `uri` equal `from`, `protocol` is a little bit more accurate.

Reply:
This PR is not to change it, and it makes some unexpected changes.